### ISSUE TITLE
question: Remove `unvalidated` feature in `imap-types`?

### ISSUE THREAD

**Issue by HenningHolmDE** [duesee/imap-codec#512]
Currently, `imap-codec` depends on the `unvalidated` feature of `imap-types`. As the crate re-exports `imap-types`, these can be used without additional opt-in:
```rust
#!/usr/bin/env rust-script
//! Create unvalidated type
//!
//! ```cargo
//! [dependencies]
//! imap-codec = { git = "https://github.com/duesee/imap-codec.git" }
//! ```

use imap_codec::imap_types::core::Text;

fn main() {
    let text = Text::unvalidated("Hello, 🦀!");
    println!("{text:?}")
}
```
```bash
$ rust-script create_unvalidated_type.ers
Text("Hello, 🦀!")
```

Is this already known to you? For me, this is not really a large problem, as the name and the documentation of the functions do provide sufficient information on how and when to use it safely.

However, does feature gating `unvalidated` in `imap-types` still make much sense? (Without it, I would not have been suprised about the re-export.)

**Comment by duesee** [comment]
Haha, nothing can hide from you :D This is great! But yes, I'm aware ...

If we look at imap-types in isolation, the feature is fine. But as soon as you implement a codec, there is a very high chance you want these functions.

In the past, `unvalidated` was marked `unsafe` and there was no `unvalidated` feature. But... I felt like both stretching the term `unsafe`, and forcing people to `#[allow(unsafe)]`.

So... the alternative was to feature-gate it. But... I don't think it makes much sense. When we say that after types comes a codec (that needs unvalidated), all higher layers will have it. This is because imap-types should only be in the dependency tree once (and featured will add up.)

Long story short: I'm open to remove the feature if we feel that there is no good use case for `imap-types` in isolation.

**Comment by duesee** [comment]
Related https://github.com/duesee/imap-codec/issues/398

**Comment by duesee** [comment]
Hello, again! I renamed this issue because there is a [similar one](https://github.com/duesee/imap-codec/issues/398) asking about the `imap-types` re-export -- I hope this was in your sense! :-) Do you have more thoughts on this? I feel that I probably want to un-gate  the `unvalidated` methods, i.e., remove the feature. Re-export is then an orthogonal question. What do you think?

**Comment by HenningHolmDE** [comment]
Removing the `unvalidated` feature gate from `imap-types` would remove the basis for my initial confusion. So yes, I'm fine with renaming the issue.
